src/Thruway/Role/Subscriber.php

```diff
@@ -128,6 +134,32 @@ protected function processSubscribed(ClientSession $session, SubscribedMessage $
             }
         }
     }
+    
+    /**
+     * Process unsubscribe error
+     * 
+     * @param \Thruway\AbstractSession $session
+     * @param \Thruway\Message\ErrorMessage $msg
+     */
+    protected function processUnsubscribeError(AbstractSession $session, ErrorMessage $msg)
+    {
+        foreach ($this->subscriptions as $key => $subscription) {
+            if ($subscription["unsubscribed_request_id"] === $msg->getErrorRequestId()) {
+                // reject the promise
+                $subscription['unsubscribed_deferred']->reject($msg);
```

Comment:
Is this deferred used anywhere? I see pieces of this idea in the code but I don't think it is used anywhere - we may be able to get rid of it.